sqlmesh/core/engine_adapter/mssql.py

```diff
@@ -185,9 +185,12 @@ def query_factory() -> Query:
             # pymssql doesn't convert Pandas Timestamp (datetime64) types
             # - this code is based on snowflake adapter implementation
             for column, kind in (columns_to_types or {}).items():
-                if kind.is_type("date") and is_datetime64_dtype(df.dtypes[column]):  # type: ignore
+                if kind.is_type("date") and is_datetime64_any_dtype(df.dtypes[column]):  # type: ignore
```

Comment:
if I am seeing this correct all of these conditions check `is_datetime64_any_dtype(df.dtypes[column])`. Although it would increase nesting, I think it would be easier to read if we did an if check on that and then did the following checks inside:
```
if is_datetime64_any_dtype(df.dtypes[column]):
   if kind.is_type("date"):
        ....
   elif getattr(df.dtypes[column], "tz", None) is not None:
        ...
   else:
      ....
```